builds/checkin/e2e-checkin.yaml

```diff
@@ -49,17 +49,24 @@ stages:
       builtImages: $[stageDependencies.CheckBuildImages.check_source_change_runtime.outputs['check_files.RUNTIMECHANGES']]
       builtPackages: $[stageDependencies.CheckBuildPackages.check_source_change_edgelet.outputs['check_files.EDGELETCHANGES']]
     jobs:
+      - template: ../e2e/templates/get-storage-uri.yaml
+        parameters:
+          azureSubscription: $(az.subscription)
+
       - job: ubuntu_2004_msmoby
         displayName: Ubuntu 20.04 with iotedge-moby
+        dependsOn: Token
+        condition: succeeded('Token')
         variables:
           verbose: false
           os: linux
           arch: amd64
           artifactName: iotedged-ubuntu20.04-amd64
           identityServiceArtifactName: packages_ubuntu-20.04_amd64
           identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+          sas_uri: $[ dependencies.Token.outputs['generate.sas_uri'] ]
         steps:
         - template: ../e2e/templates/e2e-setup.yaml
         - template: ../e2e/templates/e2e-run.yaml
-
-    
+          parameters:
+            sas_uri: $(sas_uri)
```

Comment:
You're passing a parameter named `sas_uri` here, but you didn't declare the parameter in e2e-run.yaml, and you aren't referencing `${{ parameters.sas_uri }}` inside e2e-run.yaml. Instead, you're referencing `$(sas_uri)` inside e2e-run.yaml, which happens to work because that variable was declared by the caller (above).

Can you declare the parameter at the top of the e2e-run.yaml template, and then refer to it as `${{ parameters.sas_uri }}` inside the template? Running e2e-checkin again should be all that's required to verify.